pkg/cloudprovider/providers/vsphere/vsphere_util.go

```diff
@@ -374,3 +376,118 @@ func setdatastoreFolderIDMap(
 	}
 	folderNameIDMap[folderName] = folderID
 }
+
+func convertVolPathToDevicePath(ctx context.Context, dc *vclib.Datacenter, volPath string) (string, error) {
+	volPath = vclib.RemoveClusterFromVDiskPath(volPath)
```

Comment:
Minor comment: RemoveClusterFromVDiskPath does not seem to be the right name and its quite confusing. Since we are removing the datastore/storage name from the volume path, I would suggest you to rename this method.